essential_reading/DataSources/Data_S3.md

```diff
@@ -0,0 +1,223 @@
+
+```python
+m = fs.get_mapper('s3://noaa-nwm-retrospective-2-1-zarr-pds/chrtout.zarr')
+# 'm' looks like a file as far as Python is concerned.
+# It is now a stand-in for those functions that require a file as argument.
+g = zarr.convenience.open_consolidated(m) # read zarr metadata for named file.
+print(g.tree())
+```
+
+Output:
+
+```text
+/
+ ├── crs () |S1
+ ├── elevation (2776738,) float32
+ ├── feature_id (2776738,) int32
+ ├── gage_id (2776738,) |S15
+ ├── latitude (2776738,) float32
+ ├── longitude (2776738,) float32
+ ├── order (2776738,) int32
+ ├── streamflow (367439, 2776738) int32
+ ├── time (367439,) int64
+ └── velocity (367439, 2776738) int32
+```
+
+:::{warning}
+
+We are deliberately demonstrating using commands that read **only** the zarr metadata from the
+object storage -- **not the full data set**.
+
+The above example is a very, _very_, **very** large dataset, which you don't want to load over the network to your desktop.
+Execute full data read operations only if this notebook is being hosted and run out of the same AWS center where the data lives.
+
+:::
+
+:::{sidebar}
+
+The good news about some of the larger science-oriented libraries (xarray, dask, pandas, zarr, etc), is that
+they can automatically handle the `fsspec` operations for you **IF YOUR ACCESS IS ANONYMOUS**.
+This is a convenience, but is a special case for read-only data where `anon=True` and `requestor_pays=False`.
+
+Note that this is a feature of
+[specific libraries](https://filesystem-spec.readthedocs.io/en/latest/#who-uses-fsspec),
+and doesn't work everywhere.
+
+Because it isn't universally available, and only applies to anonymous reads, examples in HyTEST will
+always explicitly plumb `fsspec` 'longhand' using `get_mapper()`. You may see example code elsewhere
+that takes the shortcut if it is available.
+
+:::
+
+
+## Credentialed Access
+
+For most data storage within the HyTEST workflows, access will not be anonymous.
+Permissions are set by the owners of that data, using credentials assigned to an AWS 'profile'.
+
+Profile credentials are usually stored outside of the Python program (typically in a master file in
+your ``HOME` folder on the compute/jupyter server).
+You need to have this set up beforehand, and is usually achieved by copying specific credential files into the right spot.
+A common mechanism to handle the profile configuration is with the `aws`
+[command line interface](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/configure/index.html).
+
+To create a new AWS profile:
+
+```sh
+> aws configure --profile osn-renci
+AWS Access Key ID :
+AWS Secret Access Key :
+Default region name : us-east-1
+Default output format: json
+```
+
+The first two prompts will ask for key/security information you were assigned for access to object storage.
+Answer last questions with "us-east-1" (region) and "json" (format).
+
+Note that this configuration is specific to the OSN 'pod' storage.
```

Comment:
Can you add a little info above about what the OSN pod is and how it relates to S3, probably near the beginning of this doc so people understand that this is demoing how to use OSN and not S3? Especially since the intro makes it sound like these are instructions for S3 (I know the process is very similar, but maybe we need a little more framing). We might also want to rename the file!

Reply:
I believe we want to say that HyTEST has procured an OSN storage pod (link) which is a ______ (something about a piece of hardware we own with object storage on a high speed network - @rsignell-usgs help me out with words here). It is similar to S3 object storage, but given that we own the hardware, we do not have to charge storage or egress fees. The OSN pod is compatible with most S3 API commands, but there are a few details that differ for reading/writing to the OSN pod, which this document will cover.

Reply:
Would also love to link in any documentation for interfacing with the pod. @rsignell-usgs may have a good link for this too.

Reply:
@amsnyder -- I am not the right person to talk about OSN's architecture, funding, ownership, who pays what egress fees, etc. Mostly because I don't know any of that information. When I have questions about what OSN is and how it works, I'd go to @rsignell-usgs ... perhaps he can blurb the information you want here.